server/cmd/mmctl/commands/compliance_export_e2e_test.go

```diff
@@ -127,3 +127,177 @@ func (s *MmctlE2ETestSuite) TestComplianceExportListCmdE2E() {
+		s.Require().EqualError(err, "failed to get compliance export job: You do not have the appropriate permissions.")
+		s.Require().Empty(printer.GetLines())
+		s.Require().Empty(printer.GetErrorLines())
+	})
+
+	s.RunForSystemAdminAndLocal("Cancel non-existent job", func(c client.Client) {
+		printer.Clean()
+
+		cmd := makeCmd()
+		err := complianceExportCancelCmdF(c, cmd, []string{"non-existent-job-id"})
+		s.Require().EqualError(err, "failed to get compliance export job: Sorry, we could not find the page., There doesn't appear to be an api call for the url='/api/v4/jobs/non-existent-job-id'.  Typo? are you missing a team_id or user_id as part of the url?")
+		s.Require().Empty(printer.GetLines())
+		s.Require().Empty(printer.GetErrorLines())
+	})
+
+	s.RunForSystemAdminAndLocal("Cancel existing job", func(c client.Client) {
+		now := model.GetMillis()
+		// Create a job
+		job, _, err := s.th.SystemAdminClient.CreateJob(context.Background(), &model.Job{
+			Id:             st.NewTestID(),
+			CreateAt:       now - 1000,
+			Status:         model.JobStatusInProgress,
+			Type:           model.JobTypeMessageExport,
+			StartAt:        now - 1000,
+			LastActivityAt: now - 1000,
+		})
+		s.Require().NoError(err)
+		defer func() {
+			// Ensure job is deleted from the database
+			var result string
+			result, err = s.th.App.Srv().Store().Job().Delete(job.Id)
+			s.Require().NoError(err, "Failed to delete job (result: %v)", result)
+		}()
+
+		printer.Clean()
+		cmd := makeCmd()
+		err = complianceExportCancelCmdF(c, cmd, []string{job.Id})
+		s.Require().NoError(err)
+		s.Require().Empty(printer.GetLines())
+		s.Require().Empty(printer.GetErrorLines())
+
+		// Verify job was cancelled
+		job, _, err = s.th.SystemAdminClient.GetJob(context.Background(), job.Id)
+		s.Require().NoError(err)
+		s.Require().Equal(model.JobStatusCanceled, job.Status)
+	})
+
+	s.RunForSystemAdminAndLocal("Error cancelling job in non-cancellable state", func(c client.Client) {
+		now := model.GetMillis()
+		// Create a job
+		job, _, err := s.th.SystemAdminClient.CreateJob(context.Background(), &model.Job{
+			Id:             st.NewTestID(),
+			CreateAt:       now - 1000,
+			Status:         model.JobStatusInProgress,
+			Type:           model.JobTypeMessageExport,
+			StartAt:        now - 1000,
+			LastActivityAt: now - 1000,
+		})
+		s.Require().NoError(err)
+		_, err = s.th.SystemAdminClient.UpdateJobStatus(context.Background(), job.Id, model.JobStatusCanceled, true)
+		s.Require().NoError(err)
+		defer func() {
+			// Ensure job is deleted from the database
+			var result string
+			result, err = s.th.App.Srv().Store().Job().Delete(job.Id)
+			s.Require().NoError(err, "Failed to delete job (result: %v)", result)
+		}()
+
+		printer.Clean()
+		cmd := makeCmd()
+		err = complianceExportCancelCmdF(c, cmd, []string{job.Id})
+		s.Require().EqualError(err, "failed to cancel compliance export job: Could not request cancellation for job that is not in a cancelable state.")
```

Comment:
Again, assuming this is the error we show to the user (not sure what difference there is between printer.GetErrorLines() and the error returned by the command function): should we specify the state the job is in?

Reply:
It is the error we return from the api. I spent some time doing some logic on the error to return something more helpful, but then realized that the error is going to be translated, so it's not worth it to detect this case. Sad. (we don't return the api error code, just the error text)